docs/design/better_coordination_multi_operator.md

```diff
@@ -0,0 +1,129 @@
+# Better coordination for multiple operator instances
+
+## Metadata
+
+* Authors: @johscheuer
+* Created: 2024-11-19
+* Updated: 2024-11-19
+
+## Background
+
+The current way to run multi-region FoundationDB deployments with the operator is to run multiple operator instances in different namespaces and/or in different Kubernetes clusters.
+This brings a few synchronization challenges as those different operator instances are not communication with each other and they are not sharing state.
+We already added the `LockClient` to synchronize the upgrade step for minor and major upgrades, where all processes must be restarted at once.
```

Comment:
There's also a general consideration we should bring into this about the difference between the upgrades and the other operations. The upgrades are built around the requirement that _every_ process in the database will be upgraded, which means that we have an external source of truth for what processes need to be ready in order for the bounce action in the upgrade to proceed. For other operations, we will have a harder time determining if the omission of processes from operator instance X indicates that there are no processes from instance X that need to be acted on, or if operator instance X has not yet processed a CRD update that would tell what processes need to be acted on. One way we can think about that is that the synchronization here is more optimistic than guaranteed, in which case we may need to be clearer about the typical expectation for the delays between different steps, and the overlap in response times between different operator instances, to know how likely it is that the signals will arrive in time to be synchronized.